src/AbstractHPolygon.jl

```diff
@@ -256,10 +258,123 @@ end
 
 # --- common AbstractHPolygon functions ---
 
+"""
+    isredundant(cmid::LinearConstraint{N}, cright::LinearConstraint{N},
+                cleft::LinearConstraint{N})::Bool where {N<:Real}
+
+Check whether a linear constraint is redundant wrt. two surrounding constraints.
+
+### Input
+
+- `cmid`   -- linear constraint of concern
+- `cright` -- linear constraint to the right (clockwise turn)
+- `cleft`  -- linear constraint to the left (counter-clockwise turn)
+
+### Output
+
+`true` iff the constraint is redundant.
+
+### Algorithm
+
+We first check whether the angle between the surrounding constraints is < 180°,
+which is a necessary condition (unless the direction is identical to one of the
+other two constraints).
+If so, we next check if the angle is 0°, in which case the constraint `cmid` is
+redundant unless it is strictly tighter than the other two constraints.
+If the angle is strictly between 0° and 180°, the constraint `cmid` is redundant
+if and only if the vertex defined by the other two constraints lies inside the
+set defined by `cmid`.
+"""
+function isredundant(cmid::LinearConstraint{N},
+                     cright::LinearConstraint{N},
+                     cleft::LinearConstraint{N})::Bool where {N<:Real}
+    samedir_check = false
+    # determine angle between surrounding constraints
+    if !is_right_turn(cright.a, cleft.a)
+        # angle is > 180°
+        samedir_check = true
+    elseif is_right_turn(cleft.a, cright.a)
+        # angle is 0° or 180°
+        if samedir(cright.a, cleft.a)[1]
+            # angle is 0°, i.e., all three constraints have the same direction
+            # constraint is redundant unless it is tighter than the other two
+            @assert samedir(cright.a, cmid.a)[1] && samedir(cleft.a, cmid.a)[1]
```

Comment:
one of these checks is redundant by transitive

Reply:
This is only a sanity check. They are always redundant.

Reply:
what i meant is that since the assertion is inside the `if samedir(cright.a, cleft.a)[1]`, one knows that this holds, hence `samedir(cright.a, cmid.a)[1]` implies that `samedir(cleft.a, cmid.a)[1]`, but ok

Reply:
Yes, I understood that.